packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -805,32 +803,27 @@ export class CrossChainMessenger {
           const withdrawalHash = hashLowLevelMessage(withdrawal)
 
           // Grab the proven withdrawal data.
-          const provenWithdrawal =
-            await this.contracts.l1.OptimismPortal2.provenWithdrawals(
-              withdrawalHash
-            )
+          const provenWithdrawal = (await this.getProvenWithdrawal(
+            withdrawalHash
+          )) as FPACProvenWithdrawal
```

Comment:
```suggestion
          const provenWithdrawal = (await this.getProvenWithdrawal(
            withdrawalHash
          ))
          if (!('proofSubmitter' in provenWithdrawal)) throw new Error("UnexpectedInternalError: Got a legacy withdrawal in a code path. This indicates a bug in the sdk")
```
Always worth doing sanity check typeguard over casting when they are pretty simple imo for discriminated unions like this

Reply:
An alternative approach is to decompose getProvenWithdrawal into getProvenLegacyWithdrawal and getProvenFpacWithdrawal. Then getProvenWithdrawal can just `if fpac: return getProvenFpacWithdrawal else getProvenLegacyWithdrawal` and if you know ahead of time you expect a fpac withdrawal you can call that method directly get the correct type returned and expect it to throw

Reply:
For the sake of mostly maintaining the old interface I'm not gonna split it up, added the sanity check.

Reply:
```suggestion
          const provenWithdrawal = (await this.getProvenWithdrawal(
            withdrawalHash
          ))
```
Can remove this typecast now that you are narrowing the type on line 811 and 821